### PR TITLE
Fix grammar multiline quote

### PR DESCRIPTION
**Pull request by vgramer**
# Description
## fix grammar: string should not be multi-line
In lexer, the closing quote for the string is optional in order to have a valid token, so the quoteHandler can autoclose it
if the user doesn't close the quote, an error will be reported by the RegoSyntaxErrorAnnotator
   

## refactor annotators to be able to test only one at the time
Introduce abstract class AnnotatorBase that adds annotation only if test mode is disabled or annotator is registered by the class under test. It allows having only the highlighting produce by the class under test in the test output

 

Fixes #

# Special notes for your reviewer

PR #73   need to be merge before in order to pass the tests. in the meantime this PR is draft

### CODE REVIEW

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.openpolicyagent.ideaplugin.ide.highlight
+
+class RegoSyntaxErrorAnnotatorTest : AnnotatorTestBase(RegoSyntaxErrorAnnotator::class){
```
Comment (anderseknert) [inline src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt:8]: Nice tests!

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    fun `test no error are reported for valid raw_string`(){
+        check_error("""
+            package main
+            a:= `hello world`
+        """.trimIndent())
+    }
+
+    fun `test no error are reported for valid multi-line raw_string`(){
```
Comment (anderseknert) [inline src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt:73]: Maybe add one for a multi-line string that does not close properly?

File: src/main/grammar/RegoLexer.flex [hunk 3]
```diff
@@ -28,10 +28,13 @@ WHITE_SPACE=\s+
 ASCII_LETTER=[A-Za-z_][A-Za-z_0-9]*
 NUMBER=-?(0|[1-9][0-9]*)(\.[0-9]+)?([eE][+-]?[0-9]*)?
 WHITE_SPACE=[ \t\n\x0B\f\r]+
-STRING_TOKEN=\"([^\"\\]|\\.)*\"
-RAW_STRING=`[^`]*`
-COMMENT=[ \t]*#[^\r\n]*
 
+//The closing quote is optional in oder to have a valid token, so the quoteHandler can autoclose it
```
Comment (anderseknert) [inline src/main/grammar/RegoLexer.flex:32]: oder -> order

**Review by anderseknert — APPROVED** [review]
Some minor remarks, but looks good.

File: src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt (hunk 2 above)
Comment (vgramer) [reply src/test/kotlin/org/openpolicyagent/ideaplugin/ide/highlight/RegoSyntaxErrorAnnotatorTest.kt:82]: good idea!  add it line 59

**Pull request merged**